src/AspNet.Security.OAuth.DingTalk/DingTalkAuthenticationHandler.cs

```diff
@@ -0,0 +1,153 @@
+            });
+
+            using var request = new HttpRequestMessage(HttpMethod.Post, address);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            request.Content = new StringContent("{\"tmp_auth_code\":\"" + tokens.AccessToken + "\"}", System.Text.Encoding.UTF8, "application/json");
+            using var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+            //using var response = await Backchannel.PostAsJsonAsync(address,new { tmp_auth_code = tokens .AccessToken});
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+            var content = await response.Content.ReadAsStringAsync();
+            using var payload = JsonDocument.Parse(content);
+
+            int status = payload.RootElement.GetProperty("errcode").GetInt32();
+            if (status != 0)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following message: {Message}.",
+                                /* Status: */ status,
+                                /* Message: */  payload.RootElement.GetString("errmsg"));
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+
+            var userInfo = payload.RootElement.GetProperty("user_info");
+            //identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, "1141CA123A8EF446A2C71DEC1BA9E2C1", ClaimValueTypes.String, Options.ClaimsIssuer));
+            var principal = new ClaimsPrincipal(identity);
+            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, userInfo);
+            context.RunClaimActions();
+
+            await Options.Events.CreatingTicket(context);
+            return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
+        }
+
+        protected override string FormatScope() => string.Join(",", Options.Scope);
+
+
+        /// <summary>
+        /// 获取时间戳
+        /// </summary>
+        /// <returns></returns>
+        private string GetTimeStamp()
+        {
+            var ts = System.DateTime.UtcNow - new System.DateTime(1970, 1, 1, 0, 0, 0, 0);
```

Comment:
Use the built-in property of type `ISystemClock` containing the clock abstraction.

Reply:
Has been adjusted